src/basic/Entity.h

```diff
@@ -71,6 +75,10 @@ class Entity {
     HitBox pickupHitBox;
     float x;
     float y;
+
+    //rendering
+    int spritePosX;
+    int spritePosY;
```

Comment:
I see no point in these, why not modify the src rect directly instead of adding an extra step?